frontend/src/app/sessions/user-sessions-wrapper/create-session/create-readonly-session/create-readonly-session.component.ts

```diff
@@ -60,33 +62,82 @@ export class CreateReadonlySessionComponent implements OnInit {
   ) {}
 
   ngOnInit(): void {
-    this.modelService.models$
-      .pipe(untilDestroyed(this), filter(Boolean))
-      .subscribe((models) => (this.models = models));
-
     this.projectService.project$
       .pipe(untilDestroyed(this), filter(Boolean))
       .subscribe((project) => (this.projectSlug = project.slug));
 
+    this.loadToolsAndModels().subscribe(([models, allVersions]) => {
+      this.resolveVersionCompatibility(models, allVersions);
+    });
+
     this.userSessionService.loadSessions();
-    this.toolService.getTools().subscribe();
+  }
+
+  loadToolsAndModels(): Observable<[Model[], ToolVersionWithTool[]]> {
+    return combineLatest([
+      this.modelService.models$.pipe(untilDestroyed(this), filter(Boolean)),
+      this.toolService.getVersionsForTools(),
+    ]).pipe(
+      tap(([_, versions]) => {
+        this.allToolVersions = versions;
+      }),
+    );
+  }
+
+  resolveVersionCompatibility(
+    models: Model[],
+    allVersions: ToolVersionWithTool[],
+  ): void {
+    this.models = [];
+
+    const modelsWithVersion = models.filter((m) => m.version);
+    for (const model of modelsWithVersion) {
```

Comment:
```suggestion
    for (const model of models) {
      if (!model.version) {
        continue;
      }
```
nit: This way we avoid using the `!` all the time when trying to access the version, it should also be more efficient, and at least for me it is as readable as the current version.